src/ios-plugin/commons/src/main/java/io/ecocode/checks/RuleCheck.java

```diff
@@ -0,0 +1,89 @@
+package io.ecocode.checks;
+
+import io.ecocode.antlr.AntlrContext;
+import io.ecocode.antlr.ParseTreeItemVisitor;
+import io.ecocode.rules.RepositoryRule;
+import io.ecocode.rules.RepositoryRuleParser;
+import org.sonar.api.batch.fs.InputFile;
+import org.sonar.api.batch.sensor.SensorContext;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.String.format;
+
+public abstract class RuleCheck implements ParseTreeItemVisitor {
+
+    private static final Logger LOGGER = Loggers.get(RuleCheck.class);
+
+    protected String ruleId;
+    protected String rulesPath;
+    protected String repositoryKey;
+
+    private class Issue {
+        protected Issue(String ruleId, int startIndex) {
+            this.ruleId = ruleId;
+            this.startIndex = startIndex;
+        }
+        protected String ruleId;
+        protected int startIndex;
+    };
+    private List<Issue> issues = new ArrayList<>();
+
+    private static List<RepositoryRule> rules = new ArrayList<>();
+    public static synchronized RepositoryRule getRepositoryRule(String ruleId, String rulesPath) throws IOException {
```

Comment:
Ok my mistake,
This is the remaining of an attempt to process multiple source files at a time... And this lock was because SensorContext is not thread safe...